config/sources/families/sun50iw9.conf

```diff
@@ -15,47 +15,65 @@ GOVERNOR=ondemand
 case $BRANCH in
 
 	legacy)
-
-		# Legacy kernel
-		LINUXFAMILY=sun50iw9
-		KERNELSOURCE='https://github.com/orangepi-xunlong/linux-orangepi.git'
-		declare -g KERNEL_MAJOR_MINOR="4.9" # Major and minor versions of this kernel.
-		KERNELBRANCH="branch:orange-pi-4.9-sun50iw9"
-		KERNELPATCHDIR=${BOARDFAMILY}-${BRANCH}
-		KERNELDIR='linux-orangepi'
-		INITRD_ARCH=arm
-		ASOUND_STATE='asound.state.sun50iw9-legacy'
-
-		# This (legacy/vendor) kernel requires uImage manual conversion; do it via a 'pre_package_kernel_image' hook.
-		declare -g NAME_KERNEL="uImage" # Even though that's a lie.
-		function pre_package_kernel_image__orangepi_legacy_uImage_manual_conversion() {
-			display_alert "Converting" "${BOARDFAMILY} - ${LINUXFAMILY} :: legacy :: uImage" "info"
-			if [[ -z "${kernel_image_pre_package_path}" || ! -f "${kernel_image_pre_package_path}" ]]; then
-				exit_with_error "kernel_image_pre_package_path ('${kernel_image_pre_package_path}') is not set or does not exist"
-			fi
-			run_host_command_logged mkimage -A arm -O linux -T kernel -C none -a "'0x40008000'" -e "'0x40008000'" -n "'Linux kernel'" -d "${kernel_image_pre_package_path}" "${kernel_image_pre_package_path}.uImage.tmp"
-			run_host_command_logged mv -v "${kernel_image_pre_package_path}.uImage.tmp" "${kernel_image_pre_package_path}"
-		}
-
-		# Legacy u-boot
-		BOOTSOURCE='https://github.com/orangepi-xunlong/u-boot-orangepi.git'
-		BOOTBRANCH='branch:v2018.05-sun50iw9'
-		BOOTPATCHDIR="legacy"
-		UBOOT_TARGET_MAP=";;dts/${BOARD}-u-boot.dts boot0_sdcard.fex boot_package.fex u-boot.bin:u-boot.fex"
-		UBOOT_COMPILER="arm-linux-gnueabi-"
-		UBOOT_USE_GCC='> 6.0'
-		BOOTENV_FILE=sun50iw9.txt
-		BOOTSCRIPT='boot-sun50iw9.cmd:boot.cmd'
-		PACK_UBOOT="yes"
-		OFFSET=20
-		ATFSOURCE=""
-		ATF_COMPILE="no"
-
-		# this overrides the one in sunxi64_common.inc
-		function write_uboot_platform() {
-			dd if=$1/boot0_sdcard.fex of=$2 bs=8k seek=1 conv=fsync
-			dd if=$1/boot_package.fex of=$2 bs=8k seek=2050 conv=fsync
-		}
+		if [[ $BOARD == bigtreetech-cb1 ]]; then
+
+			KERNELSOURCE='https://github.com/bigtreetech/linux.git'
```

Comment:
I only now noticed this. If you are going to use different source then existing for this family, you need to define new kernel family - for legacy kernel.

Reply:
Got it.

Reply:
Hi. Just define a new kernel family `sun50iw9-btt`